src/main/java/application/FormController.java

```diff
@@ -363,8 +373,9 @@ public FormEntryResponseBean getPrevious(@RequestBody SessionRequestBean request
                     formSession.getFormEntryModel(),
                     formSession.getCurrentIndex());
             updateSession(formSession, serializableFormSession);
-            FormEntryResponseBean responseBean = mapper.readValue(resp.toString(), FormEntryResponseBean.class);
+            FormEntryNavigationResponseBean responseBean = mapper.readValue(resp.toString(), FormEntryNavigationResponseBean.class);
```

Comment:
Method signature for this and getNext can be changed to FormEntryNavigationResponseBean